kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/v2/ErrorResponse.java

```diff
@@ -15,58 +15,33 @@
 
 package io.confluent.kafkarest.exceptions.v2;
 
-import static java.util.Objects.requireNonNull;
-
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import java.util.Objects;
-import java.util.StringJoiner;
+import com.google.auto.value.AutoValue;
 
 /**
  * Response returned when an exception happens in a V2 API.
  */
-public final class ErrorResponse {
-
-  private final String errorCode;
+@AutoValue
+public abstract class ErrorResponse {
```

Comment:
I don't think I understand - why do we need to keep both of the identical v2 and v3 `ErrorResponse` classes? We're far from fully isolating the v2 and the v3 implementations and even interfaces, so why the duplication?

Reply:
Oh, but we are. The V2 and V3 API are completely independent, even if they use the same implementation behind the curtains. This is so that we can evolve the V3 error response format independently of V2. It still needs to be backwards compatible to the V2 error response format, but we can add fields in V3 that won't be returned in V3 if want (and we will, for example, for the Cluster Mirror API).

Reply:
If we do strive for complete independency I think we are on very thin ice with the copious amounts of shared code (and things like the `ServerProperties.OUTBOUND_CONTENT_LENGTH_BUFFER` flag mentioned above).
This looks like a separate discussion though.